components/ac_log/android/src/test/java/mozilla/appservices/rustlog/LogTest.kt

```diff
@@ -0,0 +1,110 @@
+/* Any copyright is dedicated to the Public Domain.
+   http://creativecommons.org/publicdomain/zero/1.0/ */
+
+package mozilla.appservices.rustlog
+
+import junit.framework.Assert
+import org.junit.AfterClass
+import org.junit.BeforeClass
+import org.junit.rules.TemporaryFolder
+import org.junit.Rule
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.annotation.Config
+
+import org.junit.Test
+import org.junit.Assert.*
+import java.lang.RuntimeException
+
+@RunWith(RobolectricTestRunner::class)
+@Config(manifest = Config.NONE)
+class LogTest {
+
+    fun writeTestLog(m: String) {
+        LibRustLogAdapter.INSTANCE.ac_log_adapter_test__log_msg(m)
+        Thread.sleep(100) // Wait for it to arrive...
```

Comment:
We control everything here.  We can feed a queue (or a latch) in the log callback and access it here, avoiding the sleep entirely, _and_ ensuring the order of logging callbacks.

Reply:
Good point.

Reply:
Actually, this isn't really true. We could avoid the sleep for most cases, but not all. We need it for cases where we *don't* expect anytthing to get logged.

Reply:
Push something to the queue _after_ the log call (possibly with a very short delay).  Then check you only got that (or got it first).  It's not perfect -- there's still a race -- but you don't have slow tests outside of the one situation.